target_disk_unix.go

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:build unix
+
+package extract
+
+import (
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+// lchtimes modifies the access and modified timestamps on a target path
+// This capability is only available on unix as of now.
+func lchtimes(path string, atime, mtime time.Time) error {
+	return unix.Lutimes(path, []unix.Timeval{
+		unixTimeval(atime),
+		unixTimeval(mtime),
+	})
+}
+
+// unixTimeval converts a time.Time to a unix.Timeval. Note that it always rounds
+// up to the nearest microsecond, so even one nanosecond past the previous nanosecond
+// will be rounded up to the next microsecond.
+// See the implementation of unix.NsecToTimeval for details on how this happens.
+func unixTimeval(t time.Time) unix.Timeval {
+	return unix.NsecToTimeval(t.UnixNano())
+}
+
+// canMaintainSymlinkTimestamps determines whether is is possible to change
+// timestamps on symlinks for the the current platform. For regular files
+// and directories, attempts are made to restore permissions and timestamps
+// after extraction. But for symbolic links, go's cross-platform
+// packages (Chmod and Chtimes) are not capable of changing symlink info
+// because those methods follow the symlinks. However, a platform-dependent option
+// is provided for unix (see Lchtimes)
+func canMaintainSymlinkTimestamps() bool {
+	return true
+}
```

Comment:
Could this be replaced with a constant?

Reply:
My assumption would be that the compiler will optimize that, but happy to be explicit on that!

Reply:
However, I addressed it in https://github.com/hashicorp/go-extract/pull/106/commits/8498e88bfc0900766d4bd15cee6b3400093e7d29